src/entity/RayPiece.java

```diff
@@ -16,7 +16,7 @@ protected RayPiece(String color, int[] rayDirectionsX, int[] rayDirectionsY) {
 
```

Comment:
shouldn't RayClass be abstract? Since it's just a parent class for rook, queen, and bishop

Reply:
Good catch! I've gone and made it abstract since there's no need for it to be directly instantiated, and in fact, it shouldn't be directly instantiated, since there is other code that specifically checks that pieces are instances of Bishop, Rook, or Queen that do not know about RayPiece.